legend-engine-analytics-model-pure/src/main/resources/core_analytics_model/entityBuilder.pure

```diff
@@ -0,0 +1,389 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import meta::analytics::model::*;
+import meta::pure::mapping::*;
+import meta::relational::mapping::*;
+import meta::analytics::model::utility::*;
+import meta::relational::metamodel::*;
+
+function <<access.private>> meta::analytics::model::isPrimaryKey(pm:PropertyMapping[1], keys:Column[*]):Boolean[1]
+{
+   $pm->match([
+      r:RelationalPropertyMapping[1] | $r.relationalOperationElement->match([
+         t:TableAliasColumn[1] | $t.column->in($keys),
+         a:Any[*] | false
+      ]),
+      p:PropertyMapping[1] | false
+   ]);
+}
+
+function <<access.private>> meta::analytics::model::getPrimaryKeys(s:PropertyMappingsImplementation[0..1]):Column[*]
+{
+   $s->match([
+      e:EmbeddedRelationalInstanceSetImplementation[1] | $e.owner->getPrimaryKeys(),
+      r:RootRelationalInstanceSetImplementation[1] | $r.resolvePrimaryKey()->filter(c | $c->instanceOf(TableAliasColumn))->cast(@TableAliasColumn).column,
+      s:PropertyMappingsImplementation[0..1] | []
+   ]);
+}
+
+function <<access.private>> meta::analytics::model::getPropertyTargetType(p:PropertyMapping[1], rootClassMappings:SetImplementation[*]):Type[1]
+{
+   let targetId = $p.targetSetImplementationId;
+
+   if ($targetId->isEmpty() || $targetId == '',
+      | $p.property.genericType.rawType->toOne(),
+      |
+         $p->toOne()->match([
+            i:InlineEmbeddedRelationalInstanceSetImplementation[1] |
+               $rootClassMappings->filter(x | $x.id == $i.inlineSetImplementationId)->toOne().class;,
+            p:EmbeddedRelationalInstanceSetImplementation[1] |
+               $p.property.genericType.rawType->toOne();,
+            e:PropertyMapping[1] |
+               $rootClassMappings->filter(x | $x.id == $targetId)->toOne().class;
+   ]));
+}
+
+function <<access.private>> meta::analytics::model::getPropertyTargetName(p:PropertyMapping[1], type:Type[1], rootClassMappings:SetImplementation[*]):String[0..1]
+{
+   $p->toOne()->match([
+      i:InlineEmbeddedRelationalInstanceSetImplementation[1] |
+         let root = $rootClassMappings->filter(r | $r.id == $i.inlineSetImplementationId);
+         if ($root->isNotEmpty() && $root->toOne().root,
+             | $root->toOne().class->elementToPath(),
+             | $i.inlineSetImplementationId);,
+      p:EmbeddedRelationalInstanceSetImplementation[1] |
+         $p.targetSetImplementationId,
+      e:PropertyMapping[1] |
+         let root = $rootClassMappings->filter(r | $r.id == $p.targetSetImplementationId);
+         if ($root->isNotEmpty() && $root->toOne().root,
+             | $root->toOne().class->elementToPath(),
+             | $p.targetSetImplementationId);
+      ,
+      a:Any[1] | []
+   ])
+}
+
+function <<access.private>> meta::analytics::model::buildProperty(
+   p:AbstractProperty<Any>[1],
+   type:Type[1],
+   targetName:String[0..1],
+   isPrimaryKey: Boolean[1],
```

Comment:
I'm worried about having logic related to `relational store` here as it might make it harder for modularization, but let's see what we decide to do there